File: Leetcode/2396. Strictly Palindromic Number/solution.py
def isStrictlyPalindromic(n):
  divResult = n

  initialBase = 2
  lastBase = n - 2

  remainder = []


  while initialBase <= lastBase:
    calcRemainder = divResult % initialBase
    divResult = divResult // initialBase

    remainder.append(calcRemainder)

    if divResult == 0:
      if not isPalindrome(remainder):
        return False

      divResult = n
      initialBase += 1
      remainder = []

  return True



def isPalindrome(val):
  reversedVal = val[::-1]
  if val == reversedVal:
    return True
  else:
    return False

print(isStrictlyPalindromic(9))

File: Leetcode/2396. Strictly Palindromic Number/solution.js
// https://leetcode.com/problems/strictly-palindromic-number/

/*
  Steps to calculate the base of numbers

  1. Divide the number by the number of the base
  2. Take the remainder and save it and divide again the result by the base number
  3. repeat steps 1 and 2 until the result of division is 0
  4. Take all remainder last to first and that is the base of your number

*/

const isStrictlyPalindromic = (n) => {
  let initialBase = 2;
  const lastBase = n - 2;

  let divResult = n;

  let remainder = [];

  while (initialBase <= lastBase) {
    const remainderLoop = divResult % initialBase;
    divResult = Math.floor(divResult / initialBase);

    remainder.push(remainderLoop);

    if (divResult === 0) {
      initialBase++;
      divResult = n;

      if (!isPalindrome(remainder)) {
        return false;
      }

      remainder = [];
    }
  }

  return true;
};

const isPalindrome = (val) => {
  return val.join("") === val.reverse().join("") ? true : false;
};

console.log(isStrictlyPalindromic(9));
console.log(isStrictlyPalindromic(4));
